File: 프로그래머스/lv3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long start = 0;
        long end = 1000000000L * 1000000000L;
        while(start <= end){
            long mid = (start+end) / 2;
            long count = 0;
            for(int time : times){
                count += (mid/time);
            }
            if(count < n){
                start = mid + 1;
                answer = start;
            }
            else{
                end = mid - 1;
                answer = mid;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 1000000000 * 1000000000 + 1
    start = 0
    while start <= answer:
        temp = 0
        mid = (start+answer) // 2
        for i in range(len(times)):
            temp += mid // times[i]
        if temp >= n:
            answer = mid - 1
        else:
            start = mid + 1
    return answer + 1